### PR TITLE
Expose and wrap samtools fill_md method

### PR DESCRIPTION
**Pull request by ah19**
technically this is a private method, but it is pretty stable. The tests will fail if it changes.

There's a few unrelated gitignore/doc updates i can move out if preferred.

### CODE REVIEW

File: pysam/libcmd.pxd [hunk 1]
```diff
@@ -0,0 +1,4 @@
+from pysam.libchtslib cimport bam1_t
+
+cdef extern from "bam_md.h":
+    void bam_fillmd1_core(bam1_t *b, char *ref, int ref_len, int flag, int max_nm)
```
Comment (con-ar) [inline pysam/libcmd.pxd]: Not sure if it'll eek out much more performance, but you *might* be able to add the [`nogil`](http://cython.readthedocs.io/en/latest/src/userguide/external_C_code.html#declaring-a-function-as-callable-without-the-gil) annotation to this declaration as it looks like `bam_fillmd1_core` doesn't interact with any Python objects or have any arguments that are Python objects.

Comment (ah19) [reply pysam/libcmd.pxd]: The bam1_t struct can be modified by the AlignedSegment python object wrapper - does that make a difference? I saw nogil all over the pysam pyx files so it's presumably ok

Comment (con-ar) [reply pysam/libcmd.pxd]: I think that's okay (at least in my basic understanding of things). The worst case should just be that Cython won't compile it.

Comment (con-ar) [reply pysam/libcmd.pxd]: As you say, basically everything externed from htslib that modifies a bam1_t struct (e.g. in libchtslib.pxd) has nogil.

**Review by con-ar — APPROVED** [review]
Looks good!

File: pysam/libcfaidx.pyx [hunk 2]
```diff
@@ -245,7 +245,8 @@ cdef class FastaFile:
               reference=None,
               start=None,
               end=None,
-              region=None):
+              region=None,
+              return_bytes=None):
```
Comment (congenicajames) [inline pysam/libcfaidx.pyx]: Is there any reason why the return_bytes default value is None rather than False?

**Review by congenicajames — APPROVED** [review]

File: pysam/libcfaidx.pyx (hunk 2 above)
Comment (ah19) [reply pysam/libcfaidx.pyx]: nope, good spot!

**Pull request merged**